pkg/registry/core/rest/storage_core.go

```diff
@@ -200,7 +200,12 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(apiResourceConfigSource
 	}
 
 	serviceClusterIPAllocator, err := ipallocator.New(&serviceClusterIPRange, func(max int, rangeSpec string) (allocator.Interface, error) {
-		mem := allocator.NewAllocationMap(max, rangeSpec)
+		var mem allocator.Snapshottable
+		if utilfeature.DefaultFeatureGate.Enabled(features.ServiceIPStaticSubrange) {
+			mem = allocator.NewAllocationReservedMap(max, rangeSpec)
+		} else {
+			mem = allocator.NewAllocationMap(max, rangeSpec)
```

Comment:
@thockin now I remember, this is why I've added the logic to calculate the offset on the bitmaps , because the registry initializes the bitmap directly

Reply:
hmm, and the constructor has an interface https://github.com/kubernetes/kubernetes/blob/2374f9ad7c55999db2a6860201b835b60dfe2f3b/pkg/registry/core/service/allocator/interfaces.go#L42

Reply:
what do you think if I move this to a new constructor `ipallocator.NewInEtcd(&serviceClusterIPRange)`